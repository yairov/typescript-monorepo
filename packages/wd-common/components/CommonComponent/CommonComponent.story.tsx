import React from 'react';
import {storiesOf} from 'storybook/utils';
import {text} from '@storybook/addon-knobs/react';

import CommonComponent from './CommonComponent';

storiesOf('components', module)
  .add('CommonComponent', () => {
    const textProp = text('text', 'initial text');

    return (
      <CommonComponent text={textProp}/>
    );
  });
