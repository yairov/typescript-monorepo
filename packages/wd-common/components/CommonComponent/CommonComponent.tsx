import React from 'react';

export interface CommonComponentProps {
  text: string;
}

const CommonComponent = ({text}: CommonComponentProps) => {
  return (
    <div>
      hey! I'm a div that's being used in both projects! {text}
    </div>
  );
};

export default CommonComponent;
