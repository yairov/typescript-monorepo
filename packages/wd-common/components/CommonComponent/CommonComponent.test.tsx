import React from 'react';
import {render} from '@testing-library/react';

import CommonComponent from './CommonComponent';

describe('CommonComponent', () => {
  test('simple render', () => {
    const {asFragment} = render(<CommonComponent text="text"/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
