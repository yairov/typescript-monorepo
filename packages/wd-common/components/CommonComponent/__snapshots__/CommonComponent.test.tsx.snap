// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonComponent simple render 1`] = `
<DocumentFragment>
  <div>
    hey! I'm a div that's being used in both projects! text
  </div>
</DocumentFragment>
`;
